cvat/apps/engine/views.py

```diff
@@ -2663,47 +2662,32 @@ def get_serializer_class(self):
             return AssetWriteSerializer
 
     def create(self, request: ExtendedRequest, *args, **kwargs):
-        file = request.data.get('file', None)
+        file = request.data.get("file", None)
         if not file:
-            raise ValidationError('Asset file was not provided')
+            raise ValidationError("Asset file was not provided")
 
         if file.size / (1024 * 1024) > settings.ASSET_MAX_SIZE_MB:
-            raise ValidationError(f'Maximum size of asset is {settings.ASSET_MAX_SIZE_MB} MB')
+            raise ValidationError(f"Maximum size of asset is {settings.ASSET_MAX_SIZE_MB} MB")
 
         if file.content_type not in settings.ASSET_SUPPORTED_TYPES:
-            raise ValidationError(f'File is not supported as an asset. Supported are {settings.ASSET_SUPPORTED_TYPES}')
+            raise ValidationError(f"File is not supported as an asset. Supported are {settings.ASSET_SUPPORTED_TYPES}")
 
-        guide_id = request.data.get('guide_id')
-        db_guide = AnnotationGuide.objects.prefetch_related('assets').get(pk=guide_id)
+        guide_id = request.data.get("guide_id")
+        db_guide = AnnotationGuide.objects.prefetch_related("assets").get(pk=guide_id)
         if db_guide.assets.count() >= settings.ASSET_MAX_COUNT_PER_GUIDE:
-            raise ValidationError(f'Maximum number of assets per guide reached')
+            raise ValidationError(f"Maximum number of assets per guide reached")
 
-        serializer = self.get_serializer(data={
-            'filename': file.name,
-            'guide_id': guide_id,
+        serializer = AssetWriteSerializer(data={
```

Comment:
I might be missing something here, but why not use `data=request.data`? You'd need to rename the `asset_file` field to just `file` to match the input, but otherwise it should work. That would also make the `if not file` check unnecessary, since the serializer already checks that.

Reply:
> request.data.guide_id is a string - validation failed

Isn't that a bug in the UI, then? It shouldn't be a string in the first place.

> owner_id shouldn't be a field in the first place - I think it should, as it is a model field.

I don't think that's relevant. The purpose of the serializer is to validate input data, and `owner_id` is not part of the input. So it doesn't need to be validated and therefore doesn't need to be a serializer field.

(It's more complicated when the serializer is used for read requests, but that's not the case here.)

Reply:
> Isn't that a bug in the UI, then? It shouldn't be a string in the first place.

IDK how exactly this magic works in django, but after updating scheme it worked
I didn't touch UI, it sends multipart/form-data with integer

Reply:
Okay, nice!

Reply:
> IDK how exactly this magic works in django

FWIW, `IntegerField` does work with strings (in fact, it will cast the input value to str before casting it back to int):

https://github.com/encode/django-rest-framework/blob/33d59fefaa5af04f4bed9312239eb1e5e6def2a2/rest_framework/fields.py